src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ChimericAlignment.java

```diff
@@ -59,89 +56,231 @@
-    AlignmentRegion regionWithHigherCoordOnContig;
-    String homology;
-    String insertedSequence;
-    List<String> insertionMappings;
+    final AlignmentRegion regionWithLowerCoordOnContig;
+    final AlignmentRegion regionWithHigherCoordOnContig;
+
+    final StrandSwitch strandSwitch;
+    final boolean involvesReferenceIntervalSwitch;
+    final boolean isForwardStrandRepresentation;
+    final boolean isNotSimpleTranslocation;
+
+    final byte[] contigSeq;
+    final List<String> insertionMappings;
+
+    enum StrandSwitch {
+        NO_SWITCH, FORWARD_TO_REVERSE, REVERSE_TO_FORWARD;
+    }
+
+    // TODO: 12/14/16 Skipping simple translocation events
+    /**
+     * Parse all alignment records for a single locally-assembled contig and generate chimeric alignments if available.
+     * Applies certain filters to skip the input alignment regions that are:
+     *     1) if the alignment region's mapping quality is below a certain threshold, it is skipped
+     *     2) if the alignment region is too small, it is skipped
+     * If the alignment region passes the above two filters and the next alignment region could be treated as potential inserted sequence,
+     * note down the mapping & alignment information of that region and skip it
+     *
+     * @param input     made of ({alignmentRegions}, sequence) of a signalling locally-assembled contig
+     *
+     * @param samSequenceDictionary
+     * @return          the chimeric alignments of this sequence (empty if the sequence does not have any alignments)
+     */
+    @VisibleForTesting
+    static List<ChimericAlignment> fromSplitAlignments(final Tuple2<Iterable<AlignmentRegion>, byte[]> input,
+                                                       final SAMSequenceDictionary samSequenceDictionary) {
+
+        final byte[] sequence = input._2;
+        final List<AlignmentRegion> alignmentRegionsSortedByContigCoord = StreamSupport.stream(input._1().spliterator(), false).sorted(Comparator.comparing(a -> a.startInAssembledContig)).collect(Collectors.toList());
+        if (alignmentRegionsSortedByContigCoord.size() < 2) {
+            return new ArrayList<>();
+        }
+
+        final List<ChimericAlignment> results = new ArrayList<>(alignmentRegionsSortedByContigCoord.size() - 1);
+        final List<String> insertionAlignmentRegions = new ArrayList<>();
+
+        final Iterator<AlignmentRegion> iterator = alignmentRegionsSortedByContigCoord.iterator();
+
+        // fast forward to the first alignment region with high MapQ
+        AlignmentRegion current = iterator.next();
+        while (mapQualTooLow(current) && iterator.hasNext()) {
+            current = iterator.next();
+        }
+
+        // then iterate over the AR's in pair to identify CA's.
+        while ( iterator.hasNext() ) {
+            final AlignmentRegion next = iterator.next();
+            if (firstAlignmentIsTooShort(current, next, SVConstants.DEFAULT_MIN_ALIGNMENT_LENGTH)) {
```

Comment:
It is actually filtered out later (the code was there, I changed a bit of how the code was written in the refactor) in `nextAlignmentMayBeNovelInsertion()`.